core/templates/pages/topic-editor-page/questions-tab/topic-questions-tab.component.spec.ts

```diff
@@ -0,0 +1,224 @@
+    $rootScope = $injector.get('$rootScope');
+    $scope = $rootScope.$new();
+    $window = $injector.get('$window');
+    TopicEditorStateService = $injector.get('TopicEditorStateService');
+    TopicObjectFactory = $injector.get('TopicObjectFactory');
+    focusManagerService = $injector.get('FocusManagerService');
+    topicInitializedEventEmitter = new EventEmitter();
+    topicReinitializedEventEmitter = new EventEmitter();
+
+    ctrl = $componentController('questionsTab', {
+      $scope: $scope,
+    });
+
+    topic = TopicObjectFactory.createInterstitialTopic();
+    subtopic1 = Subtopic.createFromTitle(1, 'Subtopic1');
+    subtopic1.addSkill('skill1', 'subtopic1 skill');
+    topic.getSubtopics = function() {
+      return [subtopic1];
+    };
+
+    spyOn(TopicEditorStateService, 'getTopic').and.returnValue(topic);
+    spyOn(focusManagerService, 'setFocus');
+    spyOnProperty(TopicEditorStateService, 'onTopicInitialized').and.callFake(
+      function() {
+        return topicInitializedEventEmitter;
+      });
+    spyOnProperty(
+      TopicEditorStateService, 'onTopicReinitialized').and.callFake(
+      function() {
+        return topicReinitializedEventEmitter;
+      });
+    ctrl.$onInit();
+  }));
+
+  afterEach(() => {
+    ctrl.$onDestroy();
+  });
+
+  it('should initialize the variables', function() {
+    const topicRights = TopicRights.createInterstitialRights();
+    const allSkillSummaries = subtopic1.getSkillSummaries();
+    spyOn(TopicEditorStateService, 'getSkillIdToRubricsObject').and
+      .returnValue(skillIdToRubricsObject);
+
+    expect(focusManagerService.setFocus).toHaveBeenCalledWith(
+      'selectSkillField');
+    expect($scope.selectedSkillId).toBeNull();
+    expect($scope.question).toBeNull();
+    expect($scope.skillId).toBeNull();
+    expect($scope.topic).toBe(topic);
+    expect($scope.topicRights).toEqual(topicRights);
+    expect($scope.skillIdToRubricsObject).toEqual(skillIdToRubricsObject);
+    expect($scope.allSkillSummaries).toEqual(allSkillSummaries);
+    expect($scope.getSkillsCategorizedByTopics).toBe(CategorizedSkillsDict);
+    expect($scope.getUntriagedSkillSummaries)
+      .toBe(UntriagedSkillSummaries);
+    expect($scope.canEditQuestion).toBe(false);
+    expect($scope.misconceptions).toEqual([]);
+    expect($scope.questionIsBeingUpdated).toBe(false);
+    expect($scope.questionIsBeingSaved).toBe(false);
+    expect($scope.emptyMisconceptionsList).toEqual([]);
+  });
+
+  it('should setFocus on selectSkillField when screen loads', function() {
+    $window.onload();
+
+    expect(focusManagerService.setFocus).toHaveBeenCalledWith(
+      'selectSkillField');
+  });
+
+  it('should call initTab when topic is initialized', function() {
+    $scope.question = 'question1';
+    $scope.skillId = '1';
+    $scope.topic = null;
+
+    topicInitializedEventEmitter.emit();
+
+    expect($scope.question).toBeNull();
```

Comment:
Please add pre-checks, always check both the "before action" and "after action" states.

Also maybe add some explanation about why various fields are null? Sounds like that should be a separate test if you're focusing on behaviours (the intended behaviour this is testing is not clear from the test name, see Nithesh's comment below).

Reply:
`$scope.question` remains null and does not get updated. It is also not used in the `topic-questions-tab.component.html` file.

Reply:
OK, I'm confused. Let's go back to the test description -- everything starts from there. In what way do the various null checks test "the tab is initialized"?

Also, I still don't see the pre-checks for $scope.allSkillSummaries etc. Without those there is no proof that they were changed as a result of the action that's being tested.

Reply:
Made it a bit more clear

Reply:
Your baseline and endline verifications need to be reasonably similar to each other in order to count as baseline and endline verifications. In particular every baseline verification must have an endline counterpart.

Reply:
DONE